src/Site.js

```diff
@@ -128,7 +128,10 @@ class Site {
     this.baseUrlMap = new Set();
     this.forceReload = forceReload;
     this.plugins = {};
-    this.siteConfig = {};
+    /**
+     * @type {undefined | SiteConfig}
```

Comment:
Is it actually possible for the siteConfig to be `undefined`?

Reply:
upon initial initialization:
> `this.siteConfig = undefined;`

this is done to defend against using an empty `siteConfig`, as it is lazily initialised. ( ensure a call to `readSiteConfig` )

Since we can't have async methods in constructors, a future PR can deal with this, adopting a factory method approach to constructing `Site` that does all the io boilerplate, ensuring `siteConfig` is initialised when constructing `Site`